core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1927,4 +1927,34 @@ package object config {
       .version("3.0.1")
       .booleanConf
       .createWithDefault(false)
+
+  private[spark] val PUSH_BASED_SHUFFLE_ENABLED =
+    ConfigBuilder("spark.shuffle.push.enabled")
+      .doc("Set to 'true' to enable push based shuffle")
```

Comment:
The documentation here needs to be updated. Since we are adding a cluster level configuration in https://github.com/apache/spark/pull/30062 to enable/disable push-based shuffle, client-side configuration documentation should also reflect that enable this only if the cluster supports it.
See the comment from @Ngone51 
https://github.com/apache/spark/pull/30062#pullrequestreview-522224650

Reply:
Makes sense.

Reply:
Yes, please.

Reply:
I added the server side flag into the description.